clients/include/hipblas.hpp

```diff
@@ -28,59 +28,321 @@
+                                        int                               batch_count);
+hipblasStatus_t hipblasCcopyStridedBatchedCast(hipblasHandle_t       handle,
+                                               int                   n,
+                                               const hipblasComplex* x,
+                                               int                   incx,
+                                               hipblasStride         stridex,
+                                               hipblasComplex*       y,
+                                               int                   incy,
+                                               hipblasStride         stridey,
+                                               int                   batch_count);
+hipblasStatus_t hipblasZcopyStridedBatchedCast(hipblasHandle_t             handle,
+                                               int                         n,
+                                               const hipblasDoubleComplex* x,
+                                               int                         incx,
+                                               hipblasStride               stridex,
+                                               hipblasDoubleComplex*       y,
+                                               int                         incy,
+                                               hipblasStride               stridey,
+                                               int                         batch_count);
+
+namespace
+{
+    // Scal
+    template <typename T, typename U = T, bool FORTRAN = false>
+    hipblasStatus_t (*hipblasScal)(hipblasHandle_t handle, int n, const U* alpha, T* x, int incx);
+
+    template <typename T, typename U = T, bool FORTRAN = false>
+    hipblasStatus_t (*hipblasScalBatched)(
+        hipblasHandle_t handle, int n, const U* alpha, T* const x[], int incx, int batch_count);
+
+    template <typename T, typename U = T, bool FORTRAN = false>
+    hipblasStatus_t (*hipblasScalStridedBatched)(hipblasHandle_t handle,
+                                                 int             n,
+                                                 const U*        alpha,
+                                                 T*              x,
+                                                 int             incx,
+                                                 hipblasStride   stridex,
+                                                 int             batch_count);
+
+    MAP2CF(hipblasScal, float, float, hipblasSscal);
+    MAP2CF(hipblasScal, double, double, hipblasDscal);
+    MAP2CF_V2(hipblasScal, hipblasComplex, hipblasComplex, hipblasCscal);
+    MAP2CF_V2(hipblasScal, hipblasDoubleComplex, hipblasDoubleComplex, hipblasZscal);
+    MAP2CF_V2(hipblasScal, hipblasComplex, float, hipblasCsscal);
+    MAP2CF_V2(hipblasScal, hipblasDoubleComplex, double, hipblasZdscal);
```

Comment:
Okay so this will allow V2 with same function pointer helper?

Reply:
Yep that's the idea. The _V2 version should just add "Cast" to the end of the hipblas API, e.g. hiblasZdScal(Cast). Only to be used with complex functions.

The macros are copied directly from rocBLAS, should only have this addition.